core/src/main/java/org/apache/iceberg/actions/RewritePositionDeletesGroupPlanner.java

```diff
@@ -0,0 +1,238 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.MetadataTableType;
+import org.apache.iceberg.MetadataTableUtils;
+import org.apache.iceberg.Partitioning;
+import org.apache.iceberg.PositionDeletesScanTask;
+import org.apache.iceberg.PositionDeletesTable;
+import org.apache.iceberg.RewriteJobOrder;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.actions.RewritePositionDeleteFiles.FileGroupInfo;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.PartitionUtil;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.util.StructLikeMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Groups specified files in the {@link Table} by {@link RewriteFileGroup}s. These will be grouped
+ * by partitions. Extends the {@link SizeBasedFileRewritePlanner} with {@link
+ * RewritePositionDeleteFiles#REWRITE_JOB_ORDER} handling.
+ */
+public class RewritePositionDeletesGroupPlanner
+    extends SizeBasedFileRewritePlanner<
+        FileGroupInfo, PositionDeletesScanTask, DeleteFile, RewritePositionDeletesGroup> {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(RewritePositionDeletesGroupPlanner.class);
+
+  private final Expression filter;
+  private RewriteJobOrder rewriteJobOrder;
+
+  public RewritePositionDeletesGroupPlanner(Table table) {
+    this(table, Expressions.alwaysTrue());
+  }
+
+  /**
+   * Creates the planner for the given table.
+   *
+   * @param table to plan for
+   * @param filter used to remove files from the plan
+   */
+  public RewritePositionDeletesGroupPlanner(Table table, Expression filter) {
+    super(table);
+    this.filter = filter;
+  }
+
+  @Override
+  public Set<String> validOptions() {
+    return ImmutableSet.<String>builder()
+        .addAll(super.validOptions())
+        .add(RewritePositionDeleteFiles.REWRITE_JOB_ORDER)
```

Comment:
It is strange for me that currently we have `RewritePositionDeleteFiles.REWRITE_JOB_ORDER` and `RewriteDataFiles.REWRITE_JOB_ORDER`. If we merge these then we could move the `REWRITE_JOB_ORDER` stuff to the `SizeBasedFileRewritePlanner` class, which might be a bit nicer.